matrix_common/versionstring.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2016 OpenMarket Ltd
+# Copyright 2021-2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import functools
+import logging
+import subprocess
+from typing import Dict
+
+try:
+    from importlib.metadata import distribution
+except ImportError:
+    from importlib_metadata import distribution  # type: ignore
+
+__all__ = ["get_distribution_version_string"]
+
+logger = logging.getLogger(__name__)
+
+version_cache: Dict[str, str] = {}
+
```

Comment:
This is now unused:

```suggestion
```

(I think the `Dict` import needs to be removed too...)

Reply:
Right you are. Sorry, not sure how I missed that.

Reply:
No problem! 👍